core/src/main/java/org/apache/iceberg/view/BaseViewHistoryEntry.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface BaseViewHistoryEntry extends ViewHistoryEntry {
```

Comment:
I think we can drop `Base` since this isn't a base class. We use that when we expect something to extend the class. But this is an interface that will result in an `ImmutableViewHistoryEntry`.

Reply:
Sure let me see about a better name, the underlying interface is already ViewHistoryEntry. I think technically we could just need to add the immutable to ViewHistoryEntry but maybe its better to have an explicit core implementation which uses the annotation.

Reply:
maybe a dumb question, why don’t we just add immutable annotation on the interface in that case?

```
@Value.Immutable
public interface ViewLogEntry extends ViewHistoryEntry {
  long timestampMillis();

  int versionId();
}
```

```
public interface ViewHistoryEntry {
  /** Return the timestamp in milliseconds of the change */
  long timestampMillis();

  /** Return ID of the new current version */
  int versionId();
}
```

Reply:
yeah nvm on the classpath issue I mentioned, I confused myself. We can just add to the base interface and eliminate the duplicate interface